File: backend/src/test/java/com/pepper/backend/services/messaging/EncryptionServiceTest.java
package com.pepper.backend.services.messaging;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.security.GeneralSecurityException;
import java.util.concurrent.atomic.AtomicReference;

import static org.junit.jupiter.api.Assertions.*;

class EncryptionServiceTest {

    EncryptionService encryptionService;

    @BeforeEach
    void setUp() {
        this.encryptionService = new EncryptionService();
    }

    @Test
    void encrypt_samePassword() {
        String original = "This is a test with the same password!";

        AtomicReference<String> decrypted = new AtomicReference<>("");
        assertDoesNotThrow(() -> {
            String encrypted = this.encryptionService.encrypt(original, "pepper");
            decrypted.set(this.encryptionService.decrypt(encrypted, "pepper"));
        });
        assertEquals(original, decrypted.toString());
    }

    @Test
    void encrypt_differentPassword() {
        String original = "This is a test with different passwords!";

        AtomicReference<String> decrypted = new AtomicReference<>("");
        assertThrows(GeneralSecurityException.class, () -> {
            String encrypted = this.encryptionService.encrypt(original, "pepper");
            decrypted.set(this.encryptionService.decrypt(encrypted, "p3pper"));
        });
        assertNotEquals(original, decrypted.toString());
    }

}

File: pepper-android/care/app/src/androidTest/java/com/pepper/care/core/services/encryption/EncryptionServiceTest.kt
package com.pepper.care.core.services.encryption

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.Assert.*
import java.security.GeneralSecurityException

@RunWith(AndroidJUnit4::class)
class EncryptionServiceTest {

    val encryptionService: EncryptionService = EncryptionService()

    @Test
    fun encrypt_samePassword_success() {
        val original = "This is a test with the same password!"
        var decrypted = ""

        try {
            val encrypted = encryptionService.encrypt(original, "pepper")
            decrypted = encryptionService.decrypt(encrypted, "pepper")
        } catch (e: GeneralSecurityException) {
            e.printStackTrace()
            fail()
        }

        assertEquals(original, decrypted)
    }

    @Test
    fun encrypt_differentPassword_fails() {
        val original = "This is a test with different passwords!"
        var decrypted = ""

        try {
            val encrypted = encryptionService.encrypt(original, "pepper")
            decrypted = encryptionService.decrypt(encrypted, "p3pper")
        } catch (e: GeneralSecurityException) {
            e.printStackTrace()
            assertTrue(true)
        }

        assertNotEquals(original, decrypted)
    }

}
